profile/src/main/java/rdx/works/profile/data/repository/PublicKeyProvider.kt

```diff
@@ -26,55 +30,58 @@ class PublicKeyProvider @Inject constructor(
+            forNetworkId = forNetworkId,
+            factorSourceId = factorSource.id,
+            derivationPathScheme = DerivationPathScheme.CAP26
+        )
+        return DerivationPath.Cap26.account(
             networkId = forNetworkId,
-            accountIndex = accountIndex,
-            keyType = KeyType.TRANSACTION_SIGNING
+            keyKind = Cap26KeyKind.TRANSACTION_SIGNING,
+            index = accountIndex
         )
     }
 
     fun getDerivationPathsForIndices(
         forNetworkId: NetworkId,
-        indices: Set<Int>,
+        indices: Set<HDPathValue>,
         isForLegacyOlympia: Boolean = false
     ): List<DerivationPath> {
         return indices.map { accountIndex ->
             if (isForLegacyOlympia) {
-                DerivationPath.forLegacyOlympia(accountIndex)
+                DerivationPath.Bip44Like.init(index = accountIndex)
             } else {
-                DerivationPath.forAccount(
+                DerivationPath.Cap26.account(
                     networkId = forNetworkId,
-                    accountIndex = accountIndex,
-                    keyType = KeyType.TRANSACTION_SIGNING
+                    keyKind = Cap26KeyKind.TRANSACTION_SIGNING,
+                    index = accountIndex,
                 )
             }
         }
     }
 
     suspend fun derivePublicKeyForDeviceFactorSource(
-        deviceFactorSource: DeviceFactorSource,
+        deviceFactorSource: FactorSource.Device,
         derivationPath: DerivationPath
-    ): Result<ByteArray> {
-        return mnemonicRepository.readMnemonic(deviceFactorSource.id).mapCatching { mnemonicWithPassphrase ->
-            mnemonicWithPassphrase.compressedPublicKey(derivationPath = derivationPath).removeLeadingZero()
-        }
+    ): Result<PublicKey> = mnemonicRepository.readMnemonic(deviceFactorSource.value.id.asGeneral()).mapCatching { mnemonicWithPassphrase ->
+        mnemonicWithPassphrase.derivePublicKey(derivationPath = derivationPath)
     }
 
     suspend fun derivePublicKeysDeviceFactorSource(
-        deviceFactorSource: DeviceFactorSource,
+        deviceFactorSource: FactorSource.Device,
         derivationPaths: List<DerivationPath>,
         isForLegacyOlympia: Boolean = false
-    ): Result<Map<DerivationPath, ByteArray>> {
-        return mnemonicRepository.readMnemonic(deviceFactorSource.id).mapCatching { mnemonicWithPassphrase ->
+    ): Result<Map<DerivationPath, PublicKey>> {
+        return mnemonicRepository.readMnemonic(deviceFactorSource.value.id.asGeneral()).mapCatching { mnemonicWithPassphrase ->
             if (isForLegacyOlympia) {
                 derivationPaths.associateWith { derivationPath ->
-                    mnemonicWithPassphrase.compressedPublicKey(Slip10Curve.SECP_256K1, derivationPath)
+                    mnemonicWithPassphrase.derivePublicKey(
+                        derivationPath = derivationPath,
+                        curve = Slip10Curve.CURVE25519
```

Comment:
looks like wrong curve for legacy account

Reply:
Right this is fixed in #901 PR